cni/network/invoker_cns.go

```diff
@@ -70,94 +73,115 @@ func (invoker *CNSIPAMInvoker) Add(addConfig IPAMAddConfig) (IPAMAddResult, erro
+				return IPAMAddResult{}, errors.Wrap(errInvalidArgs, "%w: Gateway address "+info.ncGatewayIPAddress+" from response is invalid")
+			}
 
-	ncgw := net.ParseIP(info.ncGatewayIPAddress)
-	if ncgw == nil {
-		if invoker.ipamMode != util.V4Overlay {
-			return IPAMAddResult{}, errors.Wrap(errInvalidArgs, "%w: Gateway address "+info.ncGatewayIPAddress+" from response is invalid")
+			if net.ParseIP(info.podIPAddress).To4() != nil {
+				ncgw = net.ParseIP(overlayGatewayV4IP)
+			} else {
+				ncgw = net.ParseIP(overlayGatewayV6IP)
+			}
 		}
 
-		ncgw, err = getOverlayGateway(ncipnet)
-		if err != nil {
-			return IPAMAddResult{}, err
+		// set result ipconfigArgument from CNS Response Body
+		ip, ncIPNet, err := net.ParseCIDR(info.podIPAddress + "/" + fmt.Sprint(info.ncSubnetPrefix))
+		if ip == nil {
+			return IPAMAddResult{}, errors.Wrap(err, "Unable to parse IP from response: "+info.podIPAddress+" with err %w")
 		}
-	}
 
-	// construct ipnet for result
-	resultIPnet := net.IPNet{
-		IP:   ip,
-		Mask: ncipnet.Mask,
-	}
+		// construct ipnet for result
+		resultIPnet := net.IPNet{
+			IP:   ip,
+			Mask: ncIPNet.Mask,
+		}
 
-	addResult := IPAMAddResult{}
-	addResult.ipv4Result = &cniTypesCurr.Result{
-		IPs: []*cniTypesCurr.IPConfig{
-			{
-				Address: resultIPnet,
-				Gateway: ncgw,
-			},
-		},
-		Routes: []*cniTypes.Route{
-			{
-				Dst: network.Ipv4DefaultRouteDstPrefix,
-				GW:  ncgw,
-			},
-		},
-	}
+		if net.ParseIP(info.podIPAddress).To4() != nil {
+			addResult.ipv4Result = &cniTypesCurr.Result{
+				IPs: []*cniTypesCurr.IPConfig{
+					{
+						Address: resultIPnet,
+						Gateway: ncgw,
+					},
+				},
+				Routes: []*cniTypes.Route{
+					{
+						Dst: network.Ipv4DefaultRouteDstPrefix,
+						GW:  ncgw,
+					},
+				},
+			}
+		} else {
+			addResult.ipv6Result = &cniTypesCurr.Result{
+				IPs: []*cniTypesCurr.IPConfig{
+					{
+						Address: resultIPnet,
+						Gateway: ncgw,
+					},
+				},
+				Routes: []*cniTypes.Route{
+					{
+						Dst: network.Ipv6DefaultRouteDstPrefix,
+						GW:  ncgw,
+					},
+				},
+			}
+		}
 
-	// get the name of the primary IP address
-	_, hostIPNet, err := net.ParseCIDR(info.hostSubnet)
-	if err != nil {
-		return IPAMAddResult{}, fmt.Errorf("unable to parse hostSubnet: %w", err)
-	}
+		// get the name of the primary IP address
+		_, hostIPNet, err := net.ParseCIDR(info.hostSubnet)
+		if err != nil {
+			return IPAMAddResult{}, fmt.Errorf("unable to parse hostSubnet: %w", err)
+		}
 
-	addResult.hostSubnetPrefix = *hostIPNet
+		addResult.hostSubnetPrefix = *hostIPNet
 
-	// set subnet prefix for host vm
-	// setHostOptions will execute if IPAM mode is not v4 overlay
-	if invoker.ipamMode != util.V4Overlay {
-		if err := setHostOptions(ncipnet, addConfig.options, &info); err != nil {
-			return IPAMAddResult{}, err
+		// set subnet prefix for host vm
+		// setHostOptions will execute if IPAM mode is not v4 overlay
```

Comment:
Need to update this comment based on the `if` condition

Reply:
comments updated